### PR TITLE
Add tests for `publish = false` workspace members explicitly included with `--package`

### PR DESCRIPTION
**Pull request by suaviloquence**
Tests the two behaviors described in https://github.com/obi1kenobi/cargo-semver-checks/pull/859#issuecomment-2292586165.

They are currently not consistent (`--workspace --package xxx` doesn't semver-check and `--package xxx` does).

### CODE REVIEW

**Comment by suaviloquence** [comment]
Now based on #867, [compare link](https://github.com/suaviloquence/cargo-semver-checks/compare/filter-lint-totals...suaviloquence:cargo-semver-checks:test-workspace-explicitly-publish)

File: src/snapshot_tests.rs [hunk 1]
```diff
@@ -219,3 +225,45 @@ fn workspace_all_publish_false() {
...
+            "--package",
+            "a",
+        ],
+    )
+}
+
+/// Running `cargo semver-checks` on a workspace with a `package = false` member,
+/// explicitly including that member with `--package` and also specifying `--workspace`.
+/// Currently, `--workspace` overrides `--package` and the `publish = false` member is not
+/// semver-checked, and `cargo-semver-checks` silently exits 0.
```
Comment (obi1kenobi) [inline src/snapshot_tests.rs:252]: Would you mind opening an issue to make this consistent with just specifying `--package, and linking to it here?

File: src/snapshot_tests.rs [hunk 2]
```diff
@@ -219,3 +225,45 @@ fn workspace_all_publish_false() {
...
+#[test]
+fn workspace_publish_false_workspace_flag() {
+    assert_integration_test(
+        "workspace_publish_false_workspace_flag",
+        &[
+            "cargo",
+            "semver-checks",
+            "--manifest-path",
+            "test_crates/manifest_tests/workspace_all_publish_false/new",
```
Comment (obi1kenobi) [inline src/snapshot_tests.rs:261]: Another interesting question: what happens if we have `--manifest-path test_crates/manifest_tests/workspace_all_publish_false/new/a/Cargo.toml`?

Then we explicitly point to the package's manifest, ask for the workspace, then ask for the package itself.

Happy to accept this in a separate PR.

**Review by obi1kenobi — COMMENTED** [review]
Looks good, happy to merge after #867.

File: src/snapshot_tests.rs (hunk 1 above)
Comment (suaviloquence) [reply src/snapshot_tests.rs:252]: See #868.

File: src/snapshot_tests.rs [hunk 3]
```diff
@@ -243,3 +243,46 @@ fn workspace_all_publish_false() {
...
+            "--baseline-root",
+            "test_crates/manifest_tests/workspace_all_publish_false/old",
+            "--workspace",
+            "--package",
+            "a",
+            "--verbose",
```
Comment (obi1kenobi) [inline src/snapshot_tests.rs:285]:
```suggestion
            # use verbose mode to show what is being skipped
            "--verbose",
```

File: src/snapshot_tests.rs [hunk 4]
```diff
@@ -243,3 +243,46 @@ fn workspace_all_publish_false() {
...
+            "--package",
+            "a",
+        ],
+    )
+}
+
+/// Running `cargo semver-checks` on a workspace with a `package = false` member,
+/// explicitly including that member with `--package` and also specifying `--workspace`.
+/// Currently, `--workspace` overrides `--package` and the `publish = false` member is not
+/// semver-checked, and `cargo-semver-checks` silently exits 0.
```
Comment (obi1kenobi) [inline src/snapshot_tests.rs:270]:
```suggestion
/// semver-checked, and `cargo-semver-checks` silently exits 0.
///
/// Changing this behavior to make it more consistent is tracked in:
/// https://github.com/obi1kenobi/cargo-semver-checks/issues/868
```

**Review by obi1kenobi — APPROVED** [review]

File: src/snapshot_tests.rs [hunk 5]
```diff
@@ -243,3 +243,50 @@ fn workspace_all_publish_false() {
...
+            "cargo",
+            "semver-checks",
+            "--manifest-path",
+            "test_crates/manifest_tests/workspace_all_publish_false/new",
+            "--baseline-root",
+            "test_crates/manifest_tests/workspace_all_publish_false/old",
+            "--workspace",
+            "--package",
+            "a",
+            # use verbose mode to show what is being skipped
```
Comment (obi1kenobi) [inline src/snapshot_tests.rs:288]:
```suggestion
            // use verbose mode to show what is being skipped
```

**Comment by obi1kenobi** [comment]
Just added a couple of our PR review notes as source code comments, to make our thinking on the subject easier to find in future iterations.

**Pull request merged**